zqd/space/space.go

```diff
@@ -90,30 +91,39 @@ func (s Space) ConfigPath() string {
 	return filepath.Join(s.path, "config.json")
 }
 
+func (s *Space) SetPacketPath(pcapPath string) error {
+	s.conf.PacketPath = pcapPath
+	return s.conf.save(s.path)
+}
+
+func (s Space) PacketPath() string {
+	return s.conf.PacketPath
+}
+
 type config struct {
-	DataPath string `json:"data_path"`
+	DataPath   string `json:"data_path"`
+	PacketPath string `json:"packet_path"`
 }
 
-// loadConfig loads the contents of config.json. Currently DataPath is the only
-// attribute in Config, so only return the DataPath field.
-func loadConfig(name string) (string, error) {
+// loadConfig loads the contents of config.json in a space's path.
+func loadConfig(name string) (config, error) {
+	var c config
 	b, err := ioutil.ReadFile(filepath.Join(name, "config.json"))
 	if err != nil {
-		return "", err
+		return c, err
 	}
-	c := config{}
 	if err := json.Unmarshal(b, &c); err != nil {
-		return "", err
+		return c, err
 	}
-	return c.DataPath, nil
+	return c, nil
 }
 
-func saveConfig(name, dataPath string) error {
-	path := filepath.Join(name, "config.json")
+func (c config) save(spacePath string) error {
+	path := filepath.Join(spacePath, "config.json")
 	f, err := os.Create(path)
 	if err != nil {
 		return err
 	}
 	defer f.Close()
-	return json.NewEncoder(f).Encode(config{dataPath})
+	return json.NewEncoder(f).Encode(c)
```

Comment:
There's nothing to prevent two requests from corrupting this file by writing to it concurrently, but we don't need to fix that in this PR.

Reply:
Yeah, I was trying to avoid Space locks for as long as possible. I believe I can fix this will the old tmpfile rename trick.

Reply:
Agreed!